src/base/utils/fs.cpp

```diff
@@ -269,50 +269,43 @@ bool Utils::Fs::isValidFileSystemName(const QString &name, bool allowSeparators)
     return !name.contains(regex);
 }
 
-qlonglong Utils::Fs::freeDiskSpaceOnPath(QString path)
+qulonglong Utils::Fs::freeDiskSpaceOnPath(const QString &path)
 {
-    if (path.isEmpty()) return -1;
-    QDir dir_path(path);
-    if (!dir_path.exists()) {
+    if (path.isEmpty()) return 0;
+
+    QDir dirPath(path);
+    if (!dirPath.exists()) {
         QStringList parts = path.split("/");
-        while (parts.size() > 1 && !QDir(parts.join("/")).exists()) {
+        while (parts.size() > 1 && !QDir(parts.join("/")).exists())
             parts.removeLast();
-        }
-        dir_path = QDir(parts.join("/"));
-        if (!dir_path.exists()) return -1;
+
+        dirPath = QDir(parts.join("/"));
+        if (!dirPath.exists()) return 0;
     }
-    Q_ASSERT(dir_path.exists());
+    Q_ASSERT(dirPath.exists());
 
 #if defined(Q_OS_WIN)
     ULARGE_INTEGER bytesFree;
-    if (GetDiskFreeSpaceExW((LPCTSTR)(toNativePath(dir_path.path())).utf16(), &bytesFree, NULL, NULL) == 0)
-        return -1;
-    return qlonglong(bytesFree.QuadPart);
+    LPCWSTR nativePath = reinterpret_cast<LPCWSTR>((toNativePath(dirPath.path())).utf16());
+    if (GetDiskFreeSpaceExW(nativePath, &bytesFree, NULL, NULL) == 0)
+        return 0;
+    return bytesFree.QuadPart;
 #elif defined(Q_OS_HAIKU)
-    unsigned long long available;
-    const QString statfs_path = dir_path.path() + "/.";
-    dev_t device = dev_for_path (qPrintable(statfs_path));
-    if (device >= 0) {
-        fs_info info;
-        if (fs_stat_dev(device, &info) == B_OK) {
-            available = ((unsigned long long)(info.free_blocks * info.block_size));
-            return available;
-        }
-    }
-    return -1;
+    const QString statfsPath = dirPath.path() + "/.";
+    dev_t device = dev_for_path(qPrintable(statfsPath));
+    if (device < 0)
+        return 0;
+    fs_info info;
+    if (fs_stat_dev(device, &info) != B_OK)
+        return 0;
+    return ((qulonglong) info.free_blocks * info.block_size);
```

Comment:
According to [this](http://www.cplusplus.com/doc/tutorial/operators/) typecasting takes precedence from multiplication. So only `info.free_blocks` is casted to qulonglong. And we don't know the type of the result because it will also depend on the type of `info.block_size`.
So please preserve the original code aka: `return ((qulonglong)(info.free_blocks * info.block_size));`